lib/practice_exercises.rb

```diff
@@ -1,13 +1,48 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: (On)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  index = 0
+  length = list.length
+
+  while index < length do
+    while list[index] == list[index + 1] do
+      list.delete_at(index + 1)
+      length -= 1
+    end
+    index += 1
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
```

Comment:
See my note below on `strings.all?`  This is an O(n<sup>2</sup>) algorithm.  This assumes the strings are all pretty short.